crates/jxl-render/src/blend.rs

```diff
@@ -333,6 +337,80 @@ pub fn patch(
     }
 }
 
+pub fn spline(
+    image_header: &Headers,
+    base_grid: &mut [SimpleGrid<f32>],
+    spline: Spline,
+    estimated_area: u64,
+) -> crate::Result<()> {
+    // Maximum total_estimated_area_reached for Level 5
+    if estimated_area
+        > (image_header.size.height * image_header.size.width + (1 << 18)).min(1 << 22) as u64
+    {
+        tracing::warn!(
+            "Large estimated_area of splines, expect slower decoding: {}",
+            estimated_area
+        );
+    }
+    // Maximum total_estimated_area_reached for Level 10
+    // if estimated_area
+    //     > (64 * (image_header.size.height * image_header.size.width) as u64 + (1u64 << 34))
+    //         .min(1u64 << 38)
+    // {
+    //     return Err(crate::Error::Frame(
+    //         jxl_frame::Error::TooLargeEstimatedArea(estimated_area),
+    //     ));
+    // }
+
+    let all_samaples = spline.get_samples();
+    let arclength = all_samaples.len() as f32 - 2.0 + all_samaples.last().unwrap().lenght;
+    for (i, arc) in all_samaples.iter().enumerate() {
+        let arclength_from_start = f32::min(1.0, (i as f32) / arclength);
+
+        let t = 31.0 * arclength_from_start;
+        let sigma = continuous_idct(&spline.sigma_dct, t);
+        let inv_sigma = 1.0 / sigma;
+        let values = [
+            continuous_idct(&spline.xyb_dct[0], t),
+            continuous_idct(&spline.xyb_dct[1], t),
+            continuous_idct(&spline.xyb_dct[2], t),
+        ];
+
+        let max_color = f32::max(0.01, values.into_iter().reduce(f32::max).unwrap());
+        let max_distance = f32::sqrt(-2.0 * sigma * sigma * (f32::ln(0.1) * 3.0 - max_color));
```

Comment:
Maybe we can use `f32::sqrt(2.0 * (f32::ln(10.0) * 3.0 + max_color)) * sigma.abs()`? It would be more stable (as this doesn't have any subtraction,) but I might be wrong.

Reply:
Yes, that sounds correct, thanks